src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -461,20 +445,21 @@ private void StartWriting(byte[] buffer, int offset, int count, AsyncProtocolReq
                     _sslState.FinishWrite();
 
                 } while (count != 0);
+
             }
 
-            if (asyncRequest != null) 
+            if (asyncRequest != null)
             {
                 asyncRequest.CompleteUser();
             }
+        }
 
-            if (buffer == _pinnableOutputBufferInUse)
+        private void FreeBuffer(byte[] buffer)
```

Comment:
So do we want to lose the gethashcode for the logging, does seem like a waste to have for just that?